akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala

```diff
@@ -525,14 +523,13 @@ object Sink {
    * will be sent to the destination actor.
    * When the stream is completed with failure - result of `onFailureMessage(throwable)`
    * function will be sent to the destination actor.
-   *
    */
   def actorRefWithAck[T](
       ref: ActorRef,
       onInitMessage: Any,
       ackMessage: Any,
       onCompleteMessage: Any,
-      onFailureMessage: (Throwable) => Any = Status.Failure): Sink[T, NotUsed] =
+      onFailureMessage: Throwable => Any = Status.Failure): Sink[T, NotUsed] =
```

Comment:
As @johanandren pointed out this can still be a problem, so maybe we should give this one more thought before merging.

We can't change this signature without breaking binary compatibility so we will not get rid of this easily.
Only way I see that we can phase it out is to invent a new name and deprecate this one, for example `actorRefWithConfirmation` or `actorRefWithBackpressure`.

Reply:
Removing the additional brackets is a breaking binary change? (At least mima did not report anything)

Reply:
But `actorRefWithBackpressure` would anyway be a more obvious/better name than `actorRefWithAck` IMHO